doc(pe): Document policy options (#601)
armory/docs: content/en/docs/plugin-guide/plugin-policy-engine/packages/spinnaker.execution/stages.before/bake.md

---
title: "spinnaker.execution.stages.before.bake"
linktitle: "bake"

> description: "A policy targeting this object is run before executing each task in a bake stage."

Reviewer: ```suggestion
description: "A policy targeting this object runs before executing each task in a Bake stage."
```

weight: 10
---

> More information about the bake stage can be found in [Bake Amazon Machine Images in a Spinnaker Pipeline]({{< ref "aws-baking-images" >}})

Reviewer: ```suggestion
More information about the Bake stage can be found in [Bake Amazon Machine Images in a Spinnaker Pipeline]({{< ref "aws-baking-images" >}}).
```

          "name": "bindProducedArtifacts",
          "stageEnd": true,
          "stageStart": false,
          "startTime": null,
          "status": "NOT_STARTED"
        }
      ],
      "type": "bake"
    },
    "user": {
      "isAdmin": false,
      "roles": [],
      "username": "myUserName"
    }
  }
}
```

</details>

## Example Policy

Requires that baked images are of type `hvm`.

{{< prism lang="rego" line-numbers="true" >}}
package spinnaker.execution.stages.before.bake

deny["all baked images must be of type hvm"]{
	input.stage.context.vmType!="hvm"
}
{{< /prism >}}

## Keys

Parameters related to the stage against which the policy is executing can be found in the [input.stage.context](#inputstagecontext) object.

### input.pipeline

| Key                                               | Type      | Description                                                                                                                                       |
| ------------------------------------------------- | --------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `input.pipeline.application`                      | `string`  | The name of the Spinnaker application to which this pipeline belongs.                                                                             |

Reviewer: ```suggestion
| `input.pipeline.application`                      | `string`  | The name of the application to which this pipeline belongs.                                                                             |
```

| `input.pipeline.authentication.allowedAccounts[]` | `string`  | The list of accounts to which the user this stage is running as has access.                                                                       |

> | `input.pipeline.authentication.user`              | `string`  | The Spinnaker user initiating the change.                                                                                                         |

Reviewer: ```suggestion
| `input.pipeline.authentication.user`              | `string`  | The user initiating the change.                                                                                                         |
```

| `input.pipeline.buildTime`                        | `number`  |                                                                                                                                                   |
| `input.pipeline.canceled`                         | `boolean` | Has the pipeline execution been cancelled. This will always be false since the policy check requires the pipeline to be running.                  |

> | `input.pipeline.canceledBy`                       | `string ` | The userID of the user that cancelled the pipeline.                                                                                               |

Reviewer: ```suggestion
| `input.pipeline.canceledBy`                       | `string ` | The user ID of the user that cancelled the pipeline.                                                                                               |
```

| `input.pipeline.origin`                           | `string`  |                                                                                                                                                   |
| `input.pipeline.partition`                        |           |                                                                                                                                                   |
| `input.pipeline.paused`                           |           |                                                                                                                                                   |
| `input.pipeline.pipelineConfigId`                 | `string`  |                                                                                                                                                   |
| `input.pipeline.source`                           |           |                                                                                                                                                   |

> | `input.pipeline.spelEvaluator`                    | `string`  | Which version of spring expression language is being used to evaluate SpEL.                                                                       |

Reviewer: ```suggestion
| `input.pipeline.spelEvaluator`                    | `string`  | Which version of Spring Expression Language is being used to evaluate SpEL expressions.                                                                       |
```

> | `input.pipeline.stages[]`                         | `[array]` | An array of the stages in the pipeline. Typically if you are writing a policy that examines multiple pipeline stages, it is better to write that policy against either the `opa.pipelines package`, or the `spinnaker.execution.pipelines.before` package. |

Reviewer: ```suggestion
| `input.pipeline.stages[]`                         | `[array]` | An array of the stages in the pipeline. Typically if you are writing a policy that examines multiple pipeline stages, it is better to write that policy against either the `opa.pipelines package` or the `spinnaker.execution.pipelines.before` package. |
```

| `input.pipeline.startTime`                        | `number`  | Timestamp from when the pipeline was started.                                                                                                     |

> | `input.pipeline.startTimeExpiry`                  | `date `   | Unix epoch date at which the pipeline will expire.                                                                                                |

Reviewer: ```suggestion
| `input.pipeline.startTimeExpiry`                  | `date `   | Unix epoch date at which the pipeline expires.                                                                                                |
```

Reviewer: find/replace them all I reckon

| `input.pipeline.type`                             | `string`  |                                                                                                                                                   |

### input.pipeline.trigger

See [input.pipeline.trigger]({{< ref "input.pipeline.trigger.md" >}}) for more information.

### input.stage

See [`input.stage`]({{< ref "input.stage.md" >}}) for more information.

### input.stage.context

| Key                                     | Type      | Description                                                                              |
| --------------------------------------- | --------- | ---------------------------------------------------------------------------------------- |
| `input.stage.context.amiSuffix`         | `string`  | A suffix to apply to the ami. By default, a string of the date in format `YYYYMMDDHHmm`. |
| `input.stage.context.baseLabel`         | `string`  | A base label to use for the AMI.                                                         |

> | `input.stage.context.baseOs`            | `string`  | What operating system should be used to find a baseAMI to base the AMI from.             |

Reviewer: This one sounds funny
